smrt/inputs/make_bedrock.py

```diff
@@ -0,0 +1,236 @@
+"""Provides a function to build bedrock model.
+
+To create a substrate, use or implement a helper function such as `make_bedrock`. This function is able to
+automatically load a specific bedrock model and provides some bedrock permittivity formulae as well.
+
+Example::
+
+    from smrt import make_bedrock
+    bedrock = make_bedrock(substrate_model="flat", permittivity_model="christianson", bedrock_type="unfrozen bedrock", temperature=273.15)
+
+It is recommended to first carefuly read the documentation of `make_bedrock` and then explore the different types of bedrock
+models.
+"""
+
+from numbers import Number
+from typing import Callable, Union, Any, Dict, Tuple, Optional
+
+import numpy as np
+
+from smrt.core.error import SMRTError
+from smrt.core.interface import get_substrate_model
+from smrt.core.globalconstants import PERMITTIVITY_OF_FREE_SPACE 
+from smrt.core.interface import Substrate
+
+# --- 1. DATA TABLES ---
+
+# Hartlieb et al. (2016) Data: {rock_type: (real_permittivity, imaginary_permittivity)}
+# frequency = 2450 MHz, temperature = 20 degc
+HARTLIEB_2016_DATA: Dict[str, Tuple[float, float]] = {
+    'granite': (5.45, 0.038),
+    'basalt': (7.67, 0.270),
+    'sandstone': (7.67, 0.081),
+}
+
+# # Tulaczyk and Foley (2020) Data: {material_type: (real_permittivity, conductivity)}
```

Comment:
remove deadcode/commented code, or document why it must be kept (and/or not used). Necessary for the future developers.